### PR TITLE
Add AxisymmetricCenterlineAverageValue Postprocessor

### LINKED ISSUE #7528: Develop an AxisymmetricCenterlineAverage Postprocessor
### Description of the enhancement or error report
The new postprocessor will calculate the average value of a variable along axis of symmetry in an axisymmetric simulation. 

### Rationale for the enhancement or information for reproducing the error
Need the ability to output the average value of a variable along the centerline of an axisymmetric model.  

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
New feature.

### PR DESCRIPTION
**Pull request by gambka**
Closes #7528.

### CODE REVIEW

**Comment by friedmud** [comment]
Hmmm... this is required because `r=0` so `_coord[_qp]` is zero there?

That is problematic.

I guess that you're just doing the true 1D integral here... which is fine.  Why hasn't this come up before? 😄

**Comment by friedmud** [comment]
It seems like you could inherit from `SideAverageValue` and only override `computeIntegral()` and `execute()` which would reduce duplication.

That would also make more sense - this _is_ a specializetion of SideAverageValue...

**Comment by friedmud** [comment]
Could even improve things a bit by adding a `virtual Real volume()` to `SideAverageValue` so that you don't need to override `execute()` but instead just override `volume()` to do the volume calculation the way you want.

**Comment by gambka** [comment]
I am not sure why this hasn't come up before.  I will make those suggestions.

**Comment by gambka** [comment]
Can someone tell me why the one moose test would fail in threads on intel?  It looks like the expected error is given as:

Failed to access the data for variable 'u' at point

But for some reason the error printed to the screen for that particular test has an extra a at the end of data:

Failed to access the dataa for variable 'u' at point

**Comment by permcody** [comment]
This test probably shouldn't be run on threads. If both threads reach the same error condition and report it through mooseError (stderr) the output can be interleaved which causes the pattern match to fail. I would recommend that this test be restricted to a single thread/mpi process for testing purposes.

**Comment by permcody** [comment]
We have an issue related to this #7073

File: framework/src/postprocessors/AxisymmetricCenterlineAverageValue.C [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+AxisymmetricCenterlineAverageValue::AxisymmetricCenterlineAverageValue(const InputParameters & parameters) :
+    SideAverageValue(parameters),
+    _volume(0)
+{}
+
+Real
+AxisymmetricCenterlineAverageValue::volume()
+{
+  Real volume = _current_side_elem->volume();
```
Comment (friedmud) [inline framework/src/postprocessors/AxisymmetricCenterlineAverageValue.C]: No reason to save this in a local then return it - just return it!

File: framework/src/postprocessors/SideAverageValue.C [hunk 2]
```diff
@@ -48,6 +48,12 @@ SideAverageValue::getValue()
   return integral / _volume;
 }
 
+Real
+SideAverageValue::volume()
+{
+  Real volume = _current_side_volume;
```
Comment (friedmud) [inline framework/src/postprocessors/SideAverageValue.C]: Same here

**Comment by friedmud** [comment]
This is looking really good - just the couple of small issues to fixup and then I'll merge it.

Thanks for taking my advice and changing the inheritance!

**Comment by permcody** [comment]
👍

**Pull request merged**